Library/Homebrew/download_strategy.rb

```diff
@@ -269,6 +269,23 @@ def initialize(name, resource)
 
   def fetch
     ohai "Downloading #{@url}"
+
+    begin
+      does_redirect
+
+      if !does_redirect.include?(@url) && ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+        ohai "Downloaded from #{does_redirect}".gsub("Location: ", "")
+      elsif @url.include?("https") && does_redirect.include?("http")
+        puts "Insecure redirect detected! Escaping to safety."
+        exit! 1
+      end
+
+    # Don't care if it fails at this point. If it fails at this point the primary
+    # reason is that does_redirect did not return a "Location" field, i.e. the link
+    # does not silently redirect elsewhere, which is good.
+    rescue Exception => e
```

Comment:
If you use `reslove_url` I mentioned below, there shouldn't be any exception to be caught.